src/compiler/binder.ts

```diff
@@ -3340,6 +3341,9 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
                 }
             });
         }
+        if (isPrototypeProperty && namespaceSymbol && namespaceSymbol.valueDeclaration && isAssignmentDeclaration(namespaceSymbol.valueDeclaration)) {

```

Comment:
I think this check needs an explanation of how we could've gotten into this state and what this does to fix it - 2-3 sentences should do.

Reply:
something like "A prototype property assignment to something that looks class-like shouldn't actually count if the class-likeness came from an assignment declaration. The checker requires an actual class/constructor function to attach the prototype property to."